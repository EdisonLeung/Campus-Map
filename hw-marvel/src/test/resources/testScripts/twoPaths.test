# Tests pathfinding for graph with 2 paths, should find the shortest path. There are
# 3 nodes: a, b, and c. Node A will be connected to C directly and via node B.
# Pathfinding should find the shortest route from A to C (A to C directly)

# Create graph
CreateGraph twoPaths

# Add nodes
AddNode twoPaths a
AddNode twoPaths b
AddNode twoPaths c

# Add edges
AddEdge twoPaths a b AB
AddEdge twoPaths b c BC
AddEdge twoPaths a c AC

# pathfinding
FindPath twoPaths a c