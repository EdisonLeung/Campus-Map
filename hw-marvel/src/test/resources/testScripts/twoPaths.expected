# Tests pathfinding for graph with 2 paths, should find the shortest path. There are
# 3 nodes: a, b, and c. Node A will be connected to C directly and via node B.
# Pathfinding should find the shortest route from A to C (A to C directly)

# Create graph
created graph twoPaths

# Add nodes
added node a to twoPaths
added node b to twoPaths
added node c to twoPaths

# Add edges
added edge AB from a to b in twoPaths
added edge BC from b to c in twoPaths
added edge AC from a to c in twoPaths

# pathfinding
path from a to c:
a to c via AC